File: 프로그래머스/Level2/구명보트.py
# https://programmers.co.kr/learn/courses/30/lessons/42885

def solution(people, limit):
    people.sort()
    left_idx = 0
    right_idx = len(people) - 1
    answer = 0
    while left_idx <= right_idx:
        if people[left_idx] + people[right_idx] <= limit:
            answer += 1
            left_idx += 1
            right_idx -= 1
        else:
            answer += 1
            right_idx -= 1
    return answer

File: 프로그래머스/Level2/구명보트.js
// https://programmers.co.kr/learn/courses/30/lessons/42885

function solution(people, limit) {
  let answer = 0;
  people.sort((a, b) => a - b);
  let leftIdx = 0;
  let rightIdx = people.length - 1;
  while (leftIdx <= rightIdx) {
    answer++;
    if (leftIdx === rightIdx) break;
    if (people[leftIdx] + people[rightIdx] <= limit) {
      leftIdx++;
      rightIdx--;
      continue;
    } else rightIdx--;
  }
  return answer;
}
